### PR TITLE
feat: add Next Heading component

### PR DESCRIPTION
**Pull request by jacobkwan**
## Problem
Linear: https://linear.app/ogp/issue/ISOM-787/create-heading-component-in-isomer-next

Design: https://www.figma.com/file/8TEzSSDbx97gctRJ7nKdqG/Website-Template-Gen-2?type=design&node-id=1169-9443&mode=design&t=4Aw17HOEoPdVrc5z-0

Typographic component to be used whenever a typography token is used in design
variants (`lg` and `sm`) are added in as default responsive behaviour

Ideally would define custom css utility classes instead of writing code for each typography token, but not sure how to do that without overriding between themes @dcshzj

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [ ] No - this PR is backwards compatible

**Features**:

- Details ...

**Improvements**:

- Details ...

**Bug Fixes**:

- Details ...

## Before & After Screenshots

**BEFORE**:

**AFTER**:

## Tests

**New scripts**:

- `script` : script details

**New dependencies**:

- `dependency` : dependency details

**New dev dependencies**:

- `dependency` : dependency details

### CODE REVIEW

**Comment by jacobkwan** [comment]
!run chromatic

File: src/templates/next/components/Heading/Heading.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          </div>
+        )
+      })}
+    </div>
+  )
+}
+
+export const ColorsAndVariants: StoryObj<HeadingProps> = {
+  render: () => <Headings />,
+}
```
Comment (dcshzj) [inline src/templates/next/components/Heading/Heading.stories.tsx:35]: nit: quite cool but then lmao this makes it not possible to play with the props at the bottom of the storybook :(

**Review by dcshzj — APPROVED** [review]

File: src/templates/next/components/Heading/Heading.stories.tsx (hunk 1 above)
Comment (jacobkwan) [reply src/templates/next/components/Heading/Heading.stories.tsx:35]: it's still possible 😄 , I did it for Button

https://github.com/isomerpages/isomer-components/blob/main/src/templates/next/components/Button/Button.stories.tsx

Comment (dcshzj) [reply src/templates/next/components/Heading/Heading.stories.tsx:35]: I cannot get eh why ah lol. There's no box for me to change the contents.

Screenshot 2024-03-18 at 21 59 35

Comment (jacobkwan) [reply src/templates/next/components/Heading/Heading.stories.tsx:35]: oh nvm I thought you were saying that cannot add other stories with different props.
yea cannot tweak props in the rendered story

**Pull request merged**